<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<link rel="stylesheet" type="text/css" href="./perm2.css">
<title>PerM2 USC</title></head><body><div id="container">

<div id="header"><div id="logosection">
<table>
<tbody>
<tr>
<td style="padding-left:15px;">
<h1><a href="http://www-scf.usc.edu/~haifengc/perm2/"><b>PerM2</b></a></h1>
<h2><a href="http://www-scf.usc.edu/~haifengc/perm2/">DNA Sequence Mapping</a></h2>
<h2><a href="http://www.cmb.usc.edu/people/tingchen/">Ting Chen's Group</a> </h2>
<h2> <a href="http://www.cmb.usc.edu/">Computational Biology and Bioinformatics</a></h2>
<h2><a href="http://www.usc.edu">University of Southern California</a></h2>
</td>
</tr>
</tbody>
</table>
</div>
<div id="picture">
<img src="./USC.jpg" width="296" height="115"></div>
</div>

<div id="maincontent">
PerM2 is a software package for mapping DNA read sequence to genome reference sequence. The previous PerM is <a href="https://code.google.com/p/perm/">here</a>.
<h3>Source Code and Compilation</h3>
<hr SIZE=1>
<blockquote>
Download the latest version source package from <a href="PerM2SourceCode.zip">PerM2SourceCode.zip</a>. Type the following command to unzip it.<br>
<pre style="background-color: black; color: white;">$ unzip PerM2SourceCode.zip</pre>
Enter the source code folder and type the following command to compile the source code.<br>
<pre style="background-color: black; color: white;">$ make</pre>
After compiling the source code, the executable file named <b>aligner</b> will be in the source code folder.
</blockquote>
<h3>Quick Start</h3>
<hr SIZE=1>
<blockquote>
To run the command line, type <b>aligner</b> with the arguments in the order.
<pre style="background-color: black; color: white;">$ ./aligner &lt;reference&gt; &lt;reads&gt; [options]</pre>
The reference and reads file should be fasta file. The suffix of the file can be fa, fasta or other. However, if the suffix of the reference file is ".sdindex", then the reference should be an index file generated by PerM2, as the following command shown.
<pre style="background-color: black; color: white;">$ ./aligner hg19.sdindex reads.fa [options]</pre>
<b>option arguments:</b><br>
<b>-v</b>  Set the maximum number of error allowed in the reads, the default value is 2. <br>
<b>-o</b>  Set the output file name, the defalut value is the same as the read file name. <br>
<b>-s</b>  Save the reference index, the index file name is the same as the reference file name, except the suffix.
<pre style="background-color: black; color: white;">$ ./aligner hg19.fasta reads.fa -v 3 -o out.sam -s</pre>

</blockquote>
<br>
<strong>Contact Us</strong>
<hr SIZE=1>
<blockquote>
PerM2 is still under design. Any suggestions are welcomed and appreciated.<br>
<b>Haifeng Chen</b> : haifengc at usc dot edu <br>
<b>Ting Chen</b> : tingchen at usc dot edu
</blockquote>
 <br> 


</div>            

<div id="footer"><p>Last Updated on April 3, 2014 | Website maintained by <a href="http://www-scf.usc.edu/~haifengc/">Haifeng Chen</a></p>
</div>

</body></html>